packages/components/src/components/Modal/index.tsx

```diff
@@ -119,34 +170,52 @@ type ModalSubcomponents = {
 const Modal: React.FC<ModalProps> & ModalSubcomponents = ({
     children,
     heading,
+    subheading,
     header,
+    headerIcon,
+    isHeadingCentered = true,
     description,
     bottomBar,
-    cancelable = false,
+    isCancelable,
     onClick,
+    onBackClick,
     onCancel,
     totalProgressBarSteps,
     currentProgressBarStep,
     headerComponents,
     className,
     'data-test': dataTest = '@modal',
 }) => {
+    const [componentsWidth, setComponentsWidth] = useState<number>();
     const escPressed = useKeyPress('Escape');
     const theme = useTheme();
 
+    const measureComponentsRef = useCallback((element: HTMLDivElement | null) => {
```

Comment:
I am still confused with naming, if I understood correctly from rest of code, this is part of header where some header actions are renderer? Can we name this something like `headerActionsRef`, `headerActionsWidth` etc.?

Reply:
Since I made it a callback, I named it with a verb. But you are right, it's a ref for getting width of the `HeaderComponentsContainer`.